include/boost/json/monotonic_resource.hpp

```diff
@@ -18,11 +18,37 @@
 namespace boost {
 namespace json {
 
-/** A fast memory resource using many large fixed size blocks.
+/** A fast memory resource that never deallocates.
+    
+    `monotonic_resource` allocates large fixed-size
+    blocks of memory, with each allocated block being
+    twice as large as the previous block.
+    Allocated blocks are not freed until the memory
+    resource is destroyed; calls to @ref do_deallocate
+    have no effect. Memory will not be allocated unless
+    @ref do_allocate is called and the current block
```

Comment:
The user doesn't call `do_allocate` so this is confusing. The user calls `memory_resource::allocate`.